janusgraph-cassandra/src/main/java/org/janusgraph/diskstorage/cassandra/thrift/thriftpool/CTConnectionFactory.java

```diff
@@ -135,8 +143,15 @@ public boolean validateObject(String key, CTConnection c) {
                           c, curCfg, c.getConfig());
             }
         }
+        
+        boolean isOpen = c.isOpen();
+        try {
+            c.getClient().describe_version();
```

Comment:
What I mean is only call describe_version() is isOpen is true, i.e. if the connection is not open then I would assume you are always going to get the exception thrown, so it's not good to use exceptions for control flow.